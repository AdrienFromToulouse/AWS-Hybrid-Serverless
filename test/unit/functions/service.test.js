import AWS from 'aws-sdk-mock';
import LambdaTester from 'lambda-tester';

import sqsResponseQueuedMsg from '../../support/fixtures/sqsResponseQueuedMsg.json';
import ecsRunTaskResponse from '../../support/fixtures/ecsRunTaskResponse.json';
import handler from '../../../functions/service';

const payload = { attendees: 300 };

describe('service', () => {
  describe('handler()', () => {
    describe('success', () => {
      describe('returns the ECS task response', () => {
        beforeEach(() => {
          AWS.mock('SQS', 'getQueueUrl', { QueueUrl: 'QueueUrl' });
          AWS.mock('SQS', 'receiveMessage', { Messages: [payload] });
          AWS.mock('SQS', 'sendMessage', sqsResponseQueuedMsg);
          AWS.mock('SQS', 'deleteMessage');
          AWS.mock('ECS', 'runTask', ecsRunTaskResponse);
        });

        afterEach(() => {
          AWS.restore('SQS');
          AWS.restore('ECS');
        });

        test('returns result', () =>
          LambdaTester(handler.handler)
            .event(payload)
            .expectResult((data) => {
              expect(data.statusCode).toBe(200);
              expect(data.body).toMatchSnapshot();
            }),
        );
      });
    });

    describe('fail', () => {
      describe('returns error on SQS getQueueUrl', () => {
        beforeEach(() => {
          AWS.mock('SQS', 'getQueueUrl', {});
          AWS.mock('SQS', 'receiveMessage', { Messages: [payload] });
          AWS.mock('SQS', 'sendMessage', sqsResponseQueuedMsg);
          AWS.mock('SQS', 'deleteMessage');
          AWS.mock('ECS', 'runTask', ecsRunTaskResponse);
        });

        afterEach(() => {
          AWS.restore('SQS');
          AWS.restore('ECS');
        });

        test('returns result', () =>
          LambdaTester(handler.handler)
            .event(payload)
            .expectError((data) => {
              expect(data.body).toMatchSnapshot();
            }),
        );
      });
    });
  });
});
