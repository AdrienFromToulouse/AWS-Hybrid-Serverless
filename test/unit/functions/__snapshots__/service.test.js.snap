// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service handler() fail returns error on SQS getQueueUrl returns result 1`] = `undefined`;

exports[`service handler() success returns the ECS task response returns result 1`] = `
Object {
  "failures": Array [
    Object {
      "arn": "string",
      "reason": "string",
    },
  ],
  "tasks": Array [
    Object {
      "clusterArn": "string",
      "containerInstanceArn": "string",
      "containers": Array [
        Object {
          "containerArn": "string",
          "exitCode": 200,
          "lastStatus": "string",
          "name": "string",
          "reason": "string",
          "taskArn": "string",
        },
      ],
      "createdAt": 200,
      "desiredStatus": "string",
      "group": "string",
      "lastStatus": "string",
      "overrides": Object {
        "containerOverrides": Array [
          Object {
            "command": Array [
              "string",
            ],
            "environment": Array [
              Object {
                "name": "string",
                "value": "string",
              },
            ],
            "name": "string",
          },
        ],
        "taskRoleArn": "string",
      },
      "startedAt": 200,
      "startedBy": "string",
      "stoppedAt": 200,
      "stoppedReason": "string",
      "taskArn": "string",
      "taskDefinitionArn": "string",
      "version": 2,
    },
  ],
}
`;
